### DISCUSSION TITLE
Is there a concept of a canvas? Like what Adafruit GFX has?

### DISCUSSION THREAD

**Discussion by UnexpectedMaker** [Bodmer/TFT_eSPI#2952] [Q&A - General]
I know some folks have been trying to use Sprites as fun screen canvases, but I'm wondering if there is a way to create a full screen canvas (240x280) that I can draw into and then copy to the screen at a particular x,y co-ordinate?

I'm doing this now in my project with the Adafruit GFX  library to emulate sliding screens in and out based on users swiping the screen, and I'd like to move to TFT_eSPI to be able to use some of the awesome drawing elements and sprites, but I can't if there's no ability to have a separate canvas for my use case - and I can't use full screen sprites for this.

I've looked through all of the examples, but I can't see a way to do this natively in TFT_eSPI.

Thanks in advance!

**Comment by Bodmer** [comment]
In TFT_eSPI there are no "canvas" functions. You can use large sprites instead as a frame buffer (e.g. createSprite 240, 280) if your processor has sufficient RAM.
